Add docs for hipInfo and related build and tests (#5903)
ROCm/rocm-systems: projects/hip/docs/tutorial/hiptests.rst

.. meta::
  :description: The SAXPY tutorial on HIP

>   :keywords: AMD, ROCm, HIP, SAXPY, tutorial

Reviewer: Update these lines

*******************************************************************************

> Working with HIP Tests

Reviewer: ```suggestion
Working with HIP tests
```

*******************************************************************************

This tutorial introduces the HIP tests component available alongside the HIP API and runtime.
HIP tests are not installed as part of the standard ROCm installation, but you
can build ``rocm-systems/projects/hip-tests`` as described in :ref:`build-tests`.

> The catch tests and samples provided in ``hip-tests`` are both a diagnostic tool and a learning toolkit.

Reviewer: ```suggestion
The catch tests and samples provided in ``hip-tests`` serve as both a diagnostic tool and a learning toolkit.
```

> The catch tests and samples provided in ``hip-tests`` are both a diagnostic tool and a learning toolkit.

Reviewer 2: "Catch test" also mentioned in this file.

You can use the tools and tests to validate your own system environment and HIP installation. You can
also review the code and see how specific aspects of the HIP API are used under different conditions. 

Running HIP tests
=================

The HIP tests are catch tests provided alongside the HIP component, letting you test your
system and the HIP runtime. In addition, HIP tests provide some sample applications and

> utilities that can be individually built and run as needed. 

Reviewer: ```suggestion
The HIP tests are catch tests provided alongside the HIP component, allowing you to test your
system and the HIP runtime. In addition, HIP tests include sample applications and
utilities that can be built and run individually as needed. 
```

The top level of ``rocm-systems/projects/hip-tests`` is organized around one purpose: it is the official unit‑test
and sample suite for HIP. The ``rocm-systems/projects/hip-tests/catch`` folder contains the lightweight Catch2-based
C++ testing library unit test framework used by HIP tests. It provides the test harness for HIP API tests and lets
you see how HIP functions are used and validated in isolation.

> The catch tests of ``hip-tests`` serves two complementary purposes: provides a comprehensive diagnostic suite for your

Reviewer: ```suggestion
The ``hip-tests`` catch tests serve two complementary purposes: they provide a comprehensive diagnostic suite for your
```

HIP installation, and executable examples of how to write catch tests for your own code. The tests show:

* How to structure a HIP unit test
* How to wrap HIP API calls with assertions
* How to test kernels deterministically
* How to test memory behavior, events, streams, and error handling
* How to isolate failures cleanly

> * How to write reproducible GPU tests that don’t depend on timing or scheduling quirks

Reviewer: ```suggestion
HIP installation, and executable examples of how to write catch tests for your own code. The tests show how to:

* Structure a HIP unit test
* Wrap HIP API calls with assertions
* Test kernels deterministically
* Test memory behavior, events, streams, and error handling
* Isolate failures cleanly
* Write reproducible GPU tests that don’t depend on timing or scheduling quirks
```

Follow the instructions in :ref:`build-tests`, and run the tests using the ``ctest`` command from the ``build`` folder. The command runs more than 4,000 tests and writes results in the ``build/Testing/Temporary/LastTest.log`` file. You can also build and run individual tests as explained in the build instructions. 

Building and running HIP samples
================================

The ROCm/hip-tests/samples folder is organized into a set of small utilities that demonstrate how to use HIP APIs. Each sample is a utility that illustrates a concept, pattern, or technique. The folder is divided into three major groups:

* ``samples/0_Intro``: Includes basic HIP API usage and simple kernels.

  - ``samples/0_Intro/bit_extract``: Runs a small HIP kernel that extracts specific bits from integers. It demonstrates writing a small kernel, launching the kernel, managing memory, and validating results on the host.
  - ``samples/0_Intro/generic_target``: Demonstrates building code for generic GPU targets rather than for specific architectures. This is useful when you want to distribute HIP binaries that run on a variety of AMD GPUs without recompiling. 

>   - ``samples/0_Intro/module_api``: Shows how to use the HIP module API, which is analogous to CUDA’s driver API. It demonstrates loading precompiled code objects (.co or .hsaco) and launching kernels via the module API instead of the runtime API.

Reviewer 2: The "analogous to CUDA's driver API" part is a bit misleading. CUDA's driver API has more functionality than the module API; it is essentially an entirely different way to write CUDA programs. I recommend to just remove this sentence part.

```suggestion
  - ``samples/0_Intro/module_api``: Shows how to use the HIP module API. It demonstrates loading precompiled code objects (.co or .hsaco) and launching kernels via the module API instead of the runtime API.
```
